umpleonline/ump/GeometricSystem.ump

```diff
@@ -115,143 +115,154 @@ class NonPolyhedra{
-	boolean isEqual(CurvedShape object){}
+	Point[] getFoci(){return new Point[2];}
+	double getArea(){return 1.0;}
+	double getPerimeterLength(){return 1.0;}
+	Line getTangent(){return new Line(fFocus, gFocus);}
+	boolean isLessThan(CurvedShape object){return true;}
+	boolean isEqual(CurvedShape object){return true;}
 }
 class Quadrilateral{
 	isA Polygon;
-	double getArea(){/*implementation */}
-	double getPerimeterLength(){/*implementation */}
-	boolean isLessThan(Polygon object){}
-	boolean isEqual(Polygon object){}
+	double getArea(){return 1.0;}
+	double getPerimeterLength(){return 1.0;}
+	boolean isLessThan(Polygon object){return true;}
+	boolean isEqual(Polygon object){return true;}
+	boolean isEqual(CurvedShape object){return false;}
 }
 
 class Rectangle{
 	isA Quadrilateral;
 	//overwrite for better calculation
-	double getArea(){/*implementation */}
+	double getArea(){return 1.0;}
 	//overwrite for better calculation
-	double getPerimeterLength(){/*implementation */}	
+	double getPerimeterLength(){return 1.0;}	
 }
 class ArbitraryPolygon{
 	isA Polygon;
-	double getArea(){/*implementation */}
-	double getPerimeterLength(){/*implementation */}
-	boolean isLessThan(Polygon object){}
-	boolean isEqual(Polygon object){}	
+	double getArea(){return 1.0;}
+	double getPerimeterLength(){return 1.0;}
+	boolean isLessThan(Polygon object){return true;}
+	boolean isEqual(Polygon object){return true;}
+	boolean isEqual(CurvedShape object){return true;}	
 }
 
 class Sphere{
 	isA NonPolyhedra;
 	double radius;
-	double volume(){/*implementation */}
-	double surfaceArea(){/*implementation */}
-	boolean isEqual(NonPolyhedra object){/*implementation */}
-	boolean isLessThan(NonPolyhedra object){/*implementation */}
+	double volume(){return 1.0;}
+	double surfaceArea(){return 1.0;}
+	boolean isEqual(NonPolyhedra object){return true;}
+	boolean isEqual(CurvedShape object){return true;}
+	boolean isLessThan(NonPolyhedra object){return true;}
 }
 
 class Torus {
 	isA NonPolyhedra;
 	double radius_r; 
-	double radiusr_R;
+	double radius_R;
 	double theta;
 	double phi;
-	double volume(){/*implementation */}
-	double surfaceArea(){/*implementation */}
-	boolean isEqual(NonPolyhedra object){/*implementation */}
-	boolean isLessThan(NonPolyhedra object){/*implementation */}
+	double volume(){return 1.0;}
+	double surfaceArea(){return 1.0;}
+	boolean isEqual(NonPolyhedra object){return true;}
+	boolean isEqual(CurvedShape object){return true;}
+	boolean isLessThan(NonPolyhedra object){return true;}
 }
 
 class Cylinder{
 	isA NonPolyhedra;
 	double radius;
 	double height;
-	double volume(){/*implementation */}
-	double surfaceArea(){/*implementation */}	
-	boolean isEqual(NonPolyhedra object){/*implementation */}
-	boolean isLessThan(NonPolyhedra object){/*implementation */}
+	double volume(){return 1.0;}
+	double surfaceArea(){return 1.0;}	
+	boolean isEqual(NonPolyhedra object){return true;}
+	boolean isEqual(CurvedShape object){return true;}
```

Comment:
Looks like the following method should be deleted.
`boolean isEqual(CurvedShape object){return false;}`